### DISCUSSION TITLE
How to interpret explain query results?

### DISCUSSION THREAD

**Discussion by tonywang531** [pawelsalawa/sqlitestudio#3633] [Q&A]
This is a general question. When I use explain query, the results is almost always 1000 (I think that is the limit). Is this a bad thing? Most of the results are the code of "Column". Is there a way to see total number of executions for the query?

**Comment by pawelsalawa** [comment]
1000 is just one page. You can move between pages using toolbar icons (with left & right arrow icons).
Also in toolbar you can see current page number and (if you hold mouse cursor over it) total number of pages:
![image](https://user-images.githubusercontent.com/481846/72526282-5c06f180-3866-11ea-9d25-00d32b500e61.png)

You can change the limit of 1000 per page in configuration of application:

![image](https://user-images.githubusercontent.com/481846/72526311-750fa280-3866-11ea-8fc4-9b2962c37621.png)

**Comment by pawelsalawa** [comment]
To count number of rows to be returned from query, modify the query - remove any result columns from it and put count(*) instead.

**Comment by tonywang531** [comment]
> 1000 is just one page. You can move between pages using toolbar icons (with left & right arrow icons).
> Also in toolbar you can see current page number and (if you hold mouse cursor over it) total number of pages:
> ![image](https://user-images.githubusercontent.com/481846/72526282-5c06f180-3866-11ea-9d25-00d32b500e61.png)
> 
> You can change the limit of 1000 per page in configuration of application:
> 
> ![image](https://user-images.githubusercontent.com/481846/72526311-750fa280-3866-11ea-8fc4-9b2962c37621.png)

I am sorry if I did not explain clearly, when I mean "explain query" I mean this button in sqlite studio.

![image](https://user-images.githubusercontent.com/51065728/72592176-5c29ee80-3945-11ea-915c-54483b9a468c.png)

The initial setting is limited to 1000 and I cannot change page.
![image](https://user-images.githubusercontent.com/51065728/72592230-7d8ada80-3945-11ea-9e0c-f1a278280b53.png)

After changing number of data rows to 10000, I can view all the results like this: (this is the same sql query)

![image](https://user-images.githubusercontent.com/51065728/72592367-c04cb280-3945-11ea-8908-8bb3d9923da4.png)

Is this a bug by any chance? Also thanks for the count(*) tip it is a nice method which is faster than individual selection.

**Comment by pawelsalawa** [comment]
Yes, that looks like a bug.

**Comment by tonywang531** [comment]
Another similar bug:

![image](https://user-images.githubusercontent.com/51065728/72855497-0feefd80-3cfb-11ea-9ac5-a158fe3e230b.png)

The database usually fails to load the last page when the table is large (usually larger than one million). As you can see in the picture it stops somewhere strange. Refresh the table again can fix the problem. This is the "true" last page.

![image](https://user-images.githubusercontent.com/51065728/72855673-90adf980-3cfb-11ea-8a13-9a7c7482c3e6.png)

Perhaps it is better to disable view last page until it can be displayed?

**Comment by pawelsalawa** [comment]
Weird. Is there a chance you can share this db file for sake of reproducing this bug?

**Comment by tonywang531** [comment]
> Weird. Is there a chance you can share this db file for sake of reproducing this bug?

I am sorry the data is confidential and cannot be shared. I will try to create a similar database and see if I can reproduce the same bug. 

Update: I tried to replicated the same structure at home but to no anvil. I have SSD at home so even making 3 million rows the results is still displayed instantly. 

Update2: After much experimenting with the database (original), I did not find the cause of the problem. However I have found that the database not loading last page bug can be fixed by clicking "Refresh all database schemas". Perhaps this database is just too big (17G and over) and needs to refresh from time to time. I believe this solves the second bug. The first bug, remains unsolved.
